cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/BrokerFailureDetector.java

```diff
@@ -113,7 +118,15 @@ synchronized void detectBrokerFailures(boolean skipReportingIfNotUpdated) {
   }
 
   void shutdown() {
-    _zkClient.close();
+    _detectionExecutor.shutdown();
```

Comment:
Should we call `_kafkaZkClient.unregisterZNodeChildChangeHandler` with `BrokerIdsZNode.path()` during shutdown?

Reply:
In my opinion, what you suggest is more clearer than the current solution so I will add the unregister call to the source code.